### ISSUE TITLE
Request: Mono ST with next-note velocity NOT tied to 1st note.

### ISSUE THREAD

**Issue by MegsDC02** [surge-synthesizer/surge#7798]
Hello. I just came upon this thing: Surge XT is not able to control portamento time through velocity in single-trigger mono mode. Now, I have a feature request to **_make the single-trigger mono mode have separate velocity messages for each and every succeeding note_** so, for example, if the portamento time is mapped to velocity, then the **_velocity of each succeeding note in a legato line controls the portamento time_** rather than the portamento time being tied to the velocity of the first note in the legato line. I hope this feature gets considered and added into a future Surge XT release, thanks in advance!

**Comment by Andreya-Autumn** [comment]
Hello! Interesting request, thanks! I like the idea of velocity-controlled portamento. For sure this is not a change we'll make for XT1... too breaking. XT2 is the earliest it would make sense. 

There's a conversation we've been having now and then about whether Mono and Mono ST (which would be called Legato in XT2) should even be a global setting, or if there should just be Mono plus Legato switches in the relevant modulators so you can mix and match. 

Both ideas have their pros and cons, and it's too early to say which design we'll go for in the end. But if we do go for the more configurable option, the velocity modulator should indeed get the choice of retrig or legato. 

For now, I think you'll have to use the regular Mono mode for this.

**Comment by mkruselj** [comment]
In that case it should actually be release velocity of the released note IMO.

**Comment by baconpaul** [comment]
Yeah i was thinking of that.
but so few devices have accurate release velocity
we should of course reset release velocity when you release

**Comment by MegsDC02** [comment]
Hello. For example: _I play C-E-G in a legato phrase and the velocities are C:127, E:68, and G:92_, then _the portamento time if mapped to velocity currently still stays on its length at velocity 127_ — but I wish that _**the velocity of the succeeding note, in this case the E's 68 and the G's 92, controls the portamento instead**_ (and that _**there should be an option to have no portamento for any first detached note**_).

I recently made an attempt to model the brass family — including a handful of mutes — inside Surge XT (thanks to Audio Modeling's SWAM Trumpet when I demoed it, and Eric Bowing's Vital tutorial: https://youtu.be/HWAAtk6GpfM?si=X2pvFBpVHh1jvEGA), and was honestly disappointed by the fact that Surge XT doesn't have the features as I wish above, which Audio Modeling's SWAM series has (velocity controlling the attack loudness of the first note and the portamento/legato time of any succeeding note). Thanks in advance!

**Comment by baconpaul** [comment]
Yeah we get it! But it is also a breaking change and has complexities for the release - that’s what we are discussing here

**Comment by MegsDC02** [comment]
To be clear, it's the _attack or note-on_ velocity which SWAM uses for that attack (again, for the first note) and portamento (again, for any succeeding note) control in a legato line, not the _release or note-off_ velocity I'm talking about in this topic so far; the velocities 127, 68, and 92 I mentioned earlier are _attack or note-on_ velocities, not release or note-_off_ velocities. And no, it's not the release velocity I'd use for the portamento modulation in this case. TIA

**Comment by baconpaul** [comment]
Yes. We would want to set up both modulators properly if we made this change thought.

the problem is, in mono legato mode if you press c, press e, then release e, c sounds. What happens to velocity in that case..

**Comment by MegsDC02** [comment]
I'm fine setting the modulators _separately_ for the velocity and the mono/legato, so that the _legato phrase flows and the notes change smoothly with no breaks_, in this case's example _C-E-G without breaks_, but **_the velocity of the succeeding notes control the portamento time_** as per the description above and **_portamento only triggers with notes after the first note in a legato phrase_**.

Oh, and one weakness SWAM Trumpet has that Surge XT doesn't: portamento goes only up to ~300 ms for SWAM Trumpet but Surge XT's up to four seconds, which is already quite long for a real brass portamento/glissando!

I honestly hope my feature request (just like SWAM does) gets implemented at least in XT2, if not the next update! Thanks in advance...

**Comment by mkruselj** [comment]
> Edit: Also, I wish for legato too to have a crossfade time for note transitions!

Surge is not a sample library so it doesn't need this.

**Comment by MegsDC02** [comment]
> > Edit: Also, I wish for legato too to have a crossfade time for note transitions!
> 
> Surge is not a sample library so it doesn't need this.

OK... But still, I'd personally think that my feature request concerning velocity modulation in legato passages would prove _very_ useful for modeling ANY bowed and blown instrument with Surge XT!

**Comment by baconpaul** [comment]
yeah i agree having a mode where velocity adjusts via legato is a useful request.

**Comment by blancoberg** [comment]
Couldnt we add access to legato velocities via formula modulator to start with? cause that wouldnt brake anything at least. 

It would be really useful for modeling for sure

**Comment by mkruselj** [comment]
The issue is the same: we don't save the velocity of the legatoed note _in the voice manager_. So it cannot even reach the formula modulator without that plumbing.
